lightning-background-processor/src/lib.rs

```diff
@@ -263,30 +291,55 @@ mod tests {
 					(*temporary_channel_id, tx)
 				},
 				_ => panic!("Unexpected event"),
-			};
+			}
+		}}
+	}
 
-			$node_a.node.funding_transaction_generated(&temporary_channel_id, tx.clone()).unwrap();
+	macro_rules! end_open_channel {
+		($node_a: expr, $node_b: expr, $temporary_channel_id: expr, $tx: expr) => {{
+			$node_a.node.funding_transaction_generated(&$temporary_channel_id, $tx.clone()).unwrap();
 			$node_b.node.handle_funding_created(&$node_a.node.get_our_node_id(), &get_event_msg!($node_a, MessageSendEvent::SendFundingCreated, $node_b.node.get_our_node_id()));
 			$node_a.node.handle_funding_signed(&$node_b.node.get_our_node_id(), &get_event_msg!($node_b, MessageSendEvent::SendFundingSigned, $node_a.node.get_our_node_id()));
-			tx
 		}}
 	}
 
+	fn confirm_transaction(node: &mut Node, tx: &Transaction) {
+		for i in 1..=ANTI_REORG_DELAY {
+			let prev_blockhash = node.best_block.block_hash();
+			let height = node.best_block.height() + 1;
+			let header = BlockHeader { version: 0x20000000, prev_blockhash, merkle_root: Default::default(), time: height, bits: 42, nonce: 42 };
+			let txdata = vec![(0, tx)];
+			node.best_block = BestBlock::new(header.block_hash(), height);
+			match i {
+				1 => {
+					node.node.transactions_confirmed(&header, &txdata, height);
+					node.chain_monitor.transactions_confirmed(&header, &txdata, height);
+				},
+				ANTI_REORG_DELAY => {
+					node.node.best_block_updated(&header, height);
+					node.chain_monitor.best_block_updated(&header, height);
+				},
```

Comment:
Wondering why we only call this on the last block, rather than for every block in `i..=ANTI_REORG_DELAY`?